chia/full_node/full_node.py

```diff
@@ -1737,7 +1737,7 @@ async def new_compact_vdf(self, request: full_node_protocol.NewCompactVDF, peer:
         is_fully_compactified = await self.block_store.is_fully_compactified(request.header_hash)
         if is_fully_compactified is None or is_fully_compactified:
             return False
-        header_block = await self.blockchain.get_header_block_by_height(request.height, request.header_hash)
+        header_block = await self.blockchain.get_header_block_by_height(request.height, request.header_hash, True)
         if header_block is None:
             return None
```

Comment:
A note reading this:
It's very funky that those functions return both either `False` or `None`.

I know a lot of awaiting checks in the code base are performed via check
```python
if x is not None:
```
which imho is already curious, since, it's usually equivalent to
```python
if x:
```
and is used as such also for all pointed types with zero elements (`False`, `0`, `[]` and the exceptional `None`'s)

Except here where`False` and `None` can actually be distinguished.

I'm a bit wary of the `is` checks, since its python semantics have some taps. E.g., if you try in a shell
```python
>>>
>>>a = 257; b = 257; a is not b
>>>False
>>>
>>>a = 257; a is not b
>>>True
```